fix(client): remove url parameters from forum href

ShaunSHamilton opened the pull request:
- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Prevents a situation like this: https://forum.freecodecamp.org/t/problem-with-the-fcc/484974

Where I assume the behaviour comes about from:
1) Signing-in to freeCodeCamp
2) **NOT** dismissing the flash
3) Clicking any _Create a Forum Post_ button

ShaunSHamilton committed:
fix(client): remove url parameters from forum href

ShaunSHamilton committed:
add tests for removeUrlParamters

ojeytonwilliams commented on client/src/templates/Challenges/utils/index.js:
```diff
@@ -25,3 +25,7 @@ export function transformEditorLink(url) {
     );
 }
+
+export function removeUrlParameters(url) {
+  return url.replace(/\?.+/, '');
+}
```
```suggestion
export function removeUrlParameters(url) {
  return url.replace(/\?.+/, '');
}
```
Rather than using a regex, we can just use the [URL api](https://developer.mozilla.org/en-US/docs/Web/API/URL) for this. It'll be a bit more verbose, but should be more robust.

ShaunSHamilton replied:
You see, I would have thought about doing this, but I didn't...

ShaunSHamilton committed:
use URL API, Shaun's preferred method

nhcarrigan commented:
I *see* a commit that adds tests for this, but the tests don't seem to actually be in this final PR. 😕

ShaunSHamilton commented:
> I _see_ a commit that adds tests for this, but the tests don't seem to actually be in this final PR. 😕

I removed the tests, because I removed the function parsing the output.

I suppose there could be tests for this, but, as I mentioned to Oliver, I do not like mocking things, and I think it would be overkill to e2e this.

ojeytonwilliams commented:
If you want to test and not mock, you can just create a function and pass window as an argument. I.e. something that returns all the text.

Up to you, it's not a critical component.

naomi-lgbt reviewed (approved):
Works for me then!

ojeytonwilliams reviewed (approved):
Me too. @ShaunSHamilton if you're in a test writing mood, awesome. Otherwise please merge away.
